tests/test_app.py

```diff
@@ -46,6 +46,57 @@ def test_configure_logging(safe_tmpdir):
         assert sys.excepthook == excepthook
 
 
+@mock.patch('securedrop_client.app.sys.exit')
+@mock.patch('securedrop_client.app.QMessageBox')
+class TestSecondInstancePrevention(object):
+    @classmethod
+    def setup_class(cls):
+        cls.mock_app = mock.MagicMock()
+        cls.mock_app.applicationName = mock.MagicMock(return_value='sd')
+
+    @staticmethod
+    def socket_mock_generator(already_bound_errno=98):
+        namespace = set()
+
+        def kernel_bind(addr):
+            if addr in namespace:
+                error = OSError()
+                error.errno = already_bound_errno
+                raise error
+            else:
+                namespace.add(addr)
+
+        socket_mock = mock.MagicMock()
+        socket_mock.socket().bind = mock.MagicMock(side_effect=kernel_bind)
+        return socket_mock
+
+    def test_diff_name(self, mock_msgbox, mock_exit):
+        mock_socket = self.socket_mock_generator()
+        with mock.patch('securedrop_client.app.socket', new=mock_socket):
+            prevent_second_instance(self.mock_app, 'name1')
+            prevent_second_instance(self.mock_app, 'name2')
+
+            mock_exit.assert_not_called()
+
+    def test_same_name(self, mock_msgbox, mock_exit):
+        mock_socket = self.socket_mock_generator()
+        with mock.patch('securedrop_client.app.socket', new=mock_socket):
+            prevent_second_instance(self.mock_app, 'name1')
+            prevent_second_instance(self.mock_app, 'name1')
+
+            mock_exit.assert_any_call()
+
+    def test_unknown_kernel_error(self, mock_msgbox, mock_exit):
+        mock_socket = self.socket_mock_generator(131)  # crazy unexpected error
+        with mock.patch('securedrop_client.app.socket', new=mock_socket):
+            try:
```

Comment:
This should be done with `pytest.raises(OSError)` so we get a nicer error message when things go wrong.